modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionRegistry.java

```diff
@@ -103,18 +104,26 @@ ExtensionStateTracker getExtensionState(Extension extension)
         if (CollectionUtils.isEmpty(cachedCapables))
         {
             ImmutableSet.Builder<Extension> capables = ImmutableSet.builder();
-            for (Extension extension : getExtensions())
-            {
-                if (extension.isCapableOf(capabilityType))
-                {
-                    capables.add(extension);
-                }
-            }
-
+            getExtensions().stream().filter(extension -> extension.isCapableOf(capabilityType)).forEach(capables::add);
             cachedCapables = capables.build();
             capabilityToExtension.put(capabilityType, cachedCapables);
         }
 
         return cachedCapables;
     }
+
+    /**
+     * Returns a {@link Map} which values are the configuration instances which should be expired
```

Comment:
I disagree. The registration keys qualify the values. It's more natural to first enunciate a concept and then list the things that qualify it than the other way around